pkg/southbound/synchronizer/factory.go

```diff
@@ -56,9 +55,7 @@ func Factory(topoChannel <-chan *topodevice.ListResponse, opStateChan chan<- eve
 			}
 
 			device := topoEvent.Device
-			connMonitorMu.Lock()
 			connMon, ok := connectionMonitors[device.ID]
```

Comment:
I know this should be a read lock, but why doesn't this need a mutex? - i'd expect this to give a data race error

Reply:
The `connectionMonitors` map is only written and read in a loop from that one goroutine, so no lock is necessary